lib/CType.ml

```diff
@@ -100,37 +103,45 @@ let rec is_pte t = match t with
 
 let rec is_array = function
   | Array _ -> true
-  | Atomic t|Volatile t -> is_array t
-  | _ -> false
+  | Atomic t | Volatile t | Const t -> is_array t
+  | Base _ | Pointer _ -> false
 
 let rec is_atomic = function
-  | Volatile t -> is_atomic t
+  | Volatile t | Const t -> is_atomic t
   | Atomic _ -> true
-  | _ -> false
-
-let strip_atomic0 = function
-  | Atomic t -> t
-  | t -> t
+  | Array _ | Base _ | Pointer _ -> false
 
 let rec strip_atomic = function
   | Volatile t -> Volatile (strip_atomic t)
-  | t ->  strip_atomic0 t
-
-let strip_volatile0 = function
-  | Volatile t -> t
-  | t -> t
+  | Atomic t -> strip_atomic t (* We handle the case where we have malformed types *)
+  | Const t -> Const (strip_atomic t)
+  | Pointer t -> Pointer (strip_atomic t)
+  | Base s -> Base s
+  | Array (s,t) -> Array (s,t)
 
 let rec strip_volatile = function
   | Atomic t -> Atomic (strip_volatile t)
   | Pointer t -> Pointer (strip_volatile t)
-  | t ->  strip_volatile0 t
+  | Const t -> Const (strip_volatile t)
+  | Volatile t -> strip_volatile t (* We handle the case where we have malformed types*)
```

Comment:
what are malformed types, maybe that will help me understand.

Reply:
I'm not sure... I think this was an artifact of an old version of herd